pandas/core/resample.py

```diff
@@ -1197,6 +1198,8 @@ def _gotitem(self, key, ndim, subset=None):
 
         # Try to select from a DataFrame, falling back to a Series
         try:
+            if isinstance(key, list) and self.key not in key:
+                key.append(self.key)
```

Comment:
What happens when there are multiple columns in the groupby, is this section not hit? E.g.

```
rng = np.random.default_rng(0)
df_re = pd.DataFrame(
    {
        "date": pd.date_range(start="2016-01-01", periods=50),
        "group1": rng.integers(0, 2, 50),
        "group2": rng.integers(0, 2, 50),
        "val": rng.integers(0, 10, 50),
    }
)
gb = df_re.groupby(["group1", "group2"])
resample = gb.resample("10D", on="date")
subset = resample[["val"]]
res = subset.mean()
print(res)
```

Reply:
Since the subset list `["val"]` doesn't contain "date" (`self.key`), so the `append` will be hit. The result satisfies the expectation.
```
                               val
group1 group2 date
0      0      2016-01-07  0.000000
              2016-01-17  3.000000
              2016-01-27  8.000000
              2016-02-06  5.000000
              2016-02-16  1.000000
       1      2016-01-04  4.600000
              2016-01-14       NaN
              2016-01-24  6.666667
              2016-02-03  4.750000
              2016-02-13  4.000000
1      0      2016-01-02  5.000000
              2016-01-12  5.500000
              2016-01-22  5.250000
              2016-02-01  1.000000
              2016-02-11  6.500000
       1      2016-01-01  3.500000
              2016-01-11  4.750000
              2016-01-21  4.000000
              2016-01-31  9.000000
              2016-02-10  4.000000
```